Packs/PrismaSaasSecurity/Integrations/SaasSecurityEventCollector/SaasSecurityEventCollector.py

```diff
@@ -0,0 +1,254 @@
+       integration context.
+
+        Returns:
+            str: the access token.
+       """
+        integration_context = get_integration_context()
+        access_token = integration_context.get('access_token')
+        token_initiate_time = integration_context.get('token_initiate_time')
+        token_expiration_seconds = integration_context.get('token_expiration_seconds')
+
+        if access_token and not is_token_expired(
+            token_initiate_time=float(token_initiate_time), token_expiration_seconds=float(token_expiration_seconds)
+        ):
+            return access_token
+
+        # there's no token or it is expired
+        access_token, token_expiration_seconds = self.get_token_request()
+        integration_context = {
+            'access_token': access_token,
+            'token_expiration_seconds': token_expiration_seconds,
+            'token_initiate_time': time.time()
+        }
+        demisto.debug(f'updating access token - {integration_context}')
+        set_integration_context(context=integration_context)
+
+        return access_token
+
+    def get_token_request(self) -> Tuple[str, str]:
+        """
+        Sends request to retrieve token.
+
+       Returns:
+           tuple[str, str]: token and its expiration date
+        """
+        base64_encoded_creds = b64_encode(f'{self.client_id}:{self.client_secret}')
+        headers = {
+            'accept': 'application/json',
+            'Content-Type': 'application/x-www-form-urlencoded; charset=ISO-8859-1',
+            'Authorization': f'Basic {base64_encoded_creds}',
+        }
+        params = {
+            'grant_type': 'client_credentials',
+            'scope': 'api_access',
+        }
+        token_response = self._http_request('POST', url_suffix='/oauth/token', params=params, headers=headers)
+        return token_response.get('access_token'), token_response.get('expires_in')
+
+    def get_events_request(self):
+        """
+        Get up to 100 event logs.
+        """
+        return self.http_request(
+            'GET',
+            url_suffix='/api/v1/log_events_bulk',
+            resp_type='response',
+            ok_codes=[200, 204],
+        )
+
+    def get_event_request(self):
+        """
+        Get only a single event log - used only for the test-module to verify log-access api scope.
+        """
+        return self.http_request(
+            'GET',
+            url_suffix='/api/v1/log_events',
+            resp_type='response',
+            ok_codes=[200, 204],
+        )
+
+
+def is_token_expired(token_initiate_time: float, token_expiration_seconds: float) -> bool:
+    """
+    Check whether a token has expired. a token considered expired if it has been reached to its expiration date in
+    seconds minus a minute.
+
+    for example ---> time.time() = 300, token_initiate_time = 240, token_expiration_seconds = 120
+
+    300.0001 - 240 < 120 - 60
```

Comment:
basically its to be on the safe side, so for example if a token expires after 120 minutes, then on the 119 minute it will be considered as expired, not a must i can remove the -60

Reply:
Lets leave this in